zilencer/models.py

```diff
@@ -63,10 +63,11 @@ class RemotePushDeviceToken(AbstractPushDeviceToken):
 
     server: RemoteZulipServer = models.ForeignKey(RemoteZulipServer, on_delete=models.CASCADE)
     # The user id on the remote server for this device
-    user_id: int = models.BigIntegerField(db_index=True)
+    user_id: int = models.BigIntegerField(db_index=True, null=True)
+    user_uuid: UUID = models.UUIDField(null=True)
 
     class Meta:
-        unique_together = ("server", "user_id", "kind", "token")
+        unique_together = [("server", "user_id", "kind", "token"), ("user_uuid", "kind", "token")]
```

Comment:
I think this new index should have `server` in its list too; while UUIDs are unlikely to overlap between different servers, we want different servers to be treated as completely independent.